x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/AuditTrailService.java

```diff
@@ -19,219 +19,276 @@
 import java.util.Collections;
 import java.util.List;
 
-public class AuditTrailService implements AuditTrail {
+public class AuditTrailService {
 
+    private static final AuditTrail NOOP_AUDIT_TRAIL = new NoopAuditTrail();
+    private final CompositeAuditTrail compositeAuditTrail;
     private final XPackLicenseState licenseState;
-    private final List<AuditTrail> auditTrails;
-
-    @Override
-    public String name() {
-        return "service";
-    }
 
     public AuditTrailService(List<AuditTrail> auditTrails, XPackLicenseState licenseState) {
-        this.auditTrails = Collections.unmodifiableList(auditTrails);
+        this.compositeAuditTrail = new CompositeAuditTrail(Collections.unmodifiableList(auditTrails));
         this.licenseState = licenseState;
     }
 
-    /** Returns the audit trail implementations that this service delegates to. */
+    public AuditTrail get() {
+        if (compositeAuditTrail.auditTrails.isEmpty() == false && licenseState.isAuditingAllowed()) {
```

Comment:
For readability can a isEmpty method be added on the CompositeAuditTrail so that this would change to `compositeAuditTrail.isEmpty()`?